applications/photo-asset-manager/SPECIFICATION.md

```diff
@@ -80,51 +86,67 @@ A GET request is made to `/labels`. The labels are displayed to the user who can
+
+All resources share a `{NAME}` marker (a concatenation of `{PAM_NAME}-{PAM_LANG}`), chosen at AWS CDK runtime.
+Every resource is appended with a `{RANDOM}` unique ID. While the AWS CDK will truncate any generated IDs to avoid
+name lengths (such as exceeding the DNS length of 64 characters), it should be kept short and concise for easy
+legibility in the generated resource IDs. Customer must provide an `{EMAIL}` to use for account creation. This PII
+will only be used for the Amazon Cognito user pool and SNS topic.
 
-### ⭐ API Gateway
+### ⚙️ API Gateway
 
-API Gateway provides HTTP API routes for the Lambda integrations `LabelsFn`, `UploadFn`, and `PrepareDownloadFn`. Each endpoint is configured with a an Amazon Cognito authorizer. Parameters for all routes are provided in the body of the request in a JSON object. Each parameter is a top-level item in the request body JSON object.
+API Gateway provides HTTP API routes for the Lambda integrations `LabelsFn`, `UploadFn`, and `PrepareDownloadFn`. Each endpoint is configured with an Amazon Cognito authorizer. Parameters for all routes are provided in the body of the request in a JSON object. Each parameter is a top-level item in the request body JSON object.
 
 | Method | Route     | Parameters        | Example response                                             | Lambda            |
 | ------ | --------- | ----------------- | ------------------------------------------------------------ | ----------------- |
 | PUT    | /upload   | file_name: string | {"url": "presigned URL"}                                     | UploadFn          |
 | GET    | /labels   |                   | {"labels": {"maintain": {"count": 5}, "lake": {"count": 3}}} | LabelsFn          |
 | POST   | /download | labels: string[]  | {} (event)                                                   | PrepareDownloadFn |
 
-### ⭐ Amazon Cognito
+### ⚙️ Amazon Cognito
 
-A Amazon Cognito user pool is created via the AWS CDK, using the email address provided during deployment. A temporary password will be sent to that email address.
+An Amazon Cognito user pool is created via the AWS CDK, using the email address provided during deployment. A temporary password will be sent to that email address.
 
-### ⭐ S3 buckets
+### ⚙️ S3 buckets
 
 PAM uses two buckets. `{NAME}-pam-pambucketsstorage-bucket{RANDOM}` (the Storage Bucket) and `{NAME}-pam-pambucketsworking-bucket{RANDOM}` (the Working Bucket) provide the long-term Amazon S3 Intelligent-Tiering storage and ephemeral manifest and zip download storage, respectively. The AWS CDK lowercases both `{NAME}` and `{RANDOM}` to create
```

Comment:
I believe this isn't `RANDOM`, but rather `CFN Stack Deployment ID`?

Reply:
Maybe so. I'll update it when I implement in Ruby if that's true!